test/extended/util/test.go

```diff
@@ -139,6 +142,14 @@ func verifyTestSuitePreconditions() {
 		// uid don't get forced into a range (mimics upstream
 		// behavior)
 		addE2EServiceAccountsToSCC(c, namespaces, "anyuid")
+
+		// The intra-pod test requires that the service account have
+		// permission to retrieve service endpoints.
+		osClient, _, err := configapi.GetOpenShiftClient(KubeConfigPath())
+		if err != nil {
+			FatalErr(err)
+		}
+		addRoleToE2EServiceAccounts(osClient, namespaces, bootstrappolicy.EditRoleName)
```

Comment:
@marun if read access is all that is required, then let's start with the View role. For some reason, I thought I remembered you saying it needed the ability to create pods